rs/nns/governance/src/governance.rs

```diff
@@ -7016,43 +7016,11 @@ impl Governance {
         // reward weight of proposals being voted on.
         let mut actually_distributed_e8s_equivalent = 0_u64;
 
-        // Sum up "voting rights", which determine the share of the pot earned
-        // by a neuron.
-        //
-        // Construct map voters -> total _used_ voting rights for
-        // considered proposals as well as the overall total voting
-        // power on considered proposals, whether or not this voting
-        // power was used to vote (yes or no).
-        let (voters_to_used_voting_right, total_voting_rights) = {
```

Comment:
I'm not sure that moving this out of this particular function makes it clearer, or easier to understand.  

Additionally, the tests that were written were specific to using the new function, which means further refactoring would require writing new tests.  

Would it be possible to add a test that tests at a higher level, so that further changes are easier to make under test?

Reply:
Can you be more specific about what additional tests are needed? What is not already covered here?

Reply:
A test at the level of distribute_rewards for the changes made, so that we can see the whole process is correct.

I was a bit surprised no changes to existing tests were needed, but I assume it's b/c there's a default behavior with existing data structures when certain values aren't filled in.

Reply:
Would it also be possible to implement the change in the existing function, and do the refactoring / moving in a separate PR?